src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -258,6 +258,7 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 		this.currentTask = new VerifiedTask(task, resolver, trigger);
 		if (instance > 0) {
 			task.instance = this.instances[recentTaskKey].counter;
+			task._id = task._id.split('|')[0] + '|' + this.instances[recentTaskKey].counter.toString();
```

Comment:
Let's not modify the task `_id`. I've made it `readonly` to enforce this. As of 0b32b3c20ccaeeb151b875c85d59ea7eb077b23e, there's an instance property on each task.